src/Middleware/TrimMiddleware.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Engelsystem\Middleware;
+
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+
+/**
+ * Request middleware that trims all string values of PUT and POST requests.
+ * Some fields such as passwords are excluded.
+ */
+class TrimMiddleware implements MiddlewareInterface
+{
+    /**
+     * @var List of field names to exclude from trim
+     */
+    private const TRIM_EXCLUDE_LIST = ['password', 'password2'];
```

Comment:
Imho that should be set at least in a service provider (or app config?) but thats just a minor point here

Reply:
On the other hand: This is not something that should change at runtime or be configurable. It is strictly bound to the implementation. That means it is constant?

Reply:
In theory yes

Reply:
& this also needs `'new_password', 'new_password2', 'new_pw', 'new_pw2', 'password_confirmation'` added.
(which should definitely be unified on the long run to something like `'password', 'current_password', 'password_confirmation'`)

Reply:
uff²?